CondTools/HLT/plugins/AlCaRecoTriggerBitsRcdUpdate.cc

```diff
@@ -191,6 +215,86 @@ bool AlCaRecoTriggerBitsRcdUpdate::addTriggerLists(const std::vector<edm::Parame
   return true;
 }
 
+///////////////////////////////////////////////////////////////////////
+bool AlCaRecoTriggerBitsRcdUpdate::addpathsFromMap(const std::vector<edm::ParameterSet> &pathsToAdd,
+                                                   AlCaRecoTriggerBits &bits) const {
+  TriggerMap &triggerMap = bits.m_alcarecoToTrig;
+
+  // loop on PSets, each containing the key (filter name) and a vstring with triggers
+  for (std::vector<edm::ParameterSet>::const_iterator iSet = pathsToAdd.begin(); iSet != pathsToAdd.end(); ++iSet) {
+    const std::string filter(iSet->getParameter<std::string>("listName"));
+    std::string mergedPathsInKey;
+
+    for (const auto &imap : triggerMap) {
+      if (imap.first == filter)
+        mergedPathsInKey = imap.second;
+    }
+
+    std::vector<std::string> PathsInKey = bits.decompose(mergedPathsInKey);
+    std::vector<std::string> PathsInKeyTmp = PathsInKey;
+    std::vector<std::string> paths(iSet->getParameter<std::vector<std::string>>("hltPaths"));
+
+    for (const auto &ipath : paths) {
+      bool isPathsInKey = false;
+      for (auto it = PathsInKey.begin(); it != PathsInKey.end();) {
+        if ((*it) == ipath) {
+          isPathsInKey = true;
+          break;
+        } else
+          ++it;
+      }
+      if (isPathsInKey == false)
+        PathsInKeyTmp.push_back(ipath);
+    }
```

Comment:
If the intent is to merge the two vectors removing duplicates, one could use `std::set` to make the impl more compact (even though it is still necessary to convert back to `std::vector` to feed `bits.compose`).
```suggestion
    auto const& pathsInKey = bits.decompose(mergedPathsInKey);
    auto const& paths = iSet->getParameter<std::vector<std::string>>("hltPaths");
    std::set<std::string> pathsSet{pathsInKey.begin(), pathsInKey.end()};
    std::copy(paths.begin(), paths.end(), std::inserter(pathsSet, pathsSet.end()));
    std::vector<std::string> const newPathsInKey{pathsSet.begin(), pathsSet.end()};
```
Note-1: with this alternative impl, possible duplicates in `PathsInKey` are also removed (which the current impl does not do, maybe for a good reason that I don't know).
Note-2: I didn't test this code, nor compared impls in terms of CPU performance (but I guess we are dealing with relatively small vectors here).